DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -88,12 +93,26 @@ IUserDataService userDataService
 
             passwordDataService.SetPasswordByCandidateNumber(candidateNumber, delegateRegistrationModel.PasswordHash!);
 
-            UpdateSupervisorDelegateRecordsWithNewCandidateId(
+            var delegateUser = userDataService.GetDelegateUserByCandidateNumber(
                 candidateNumber,
-                delegateRegistrationModel.Centre,
-                supervisorDelegateRecordIdsMatchingDelegate
+                delegateRegistrationModel.Centre
             );
 
+            if (delegateUser == null)

```

Comment:
I don't see how we can hit this case without some error with the code above? The stored procedure to register the delegate can only return a valid non-null candidateNumber, "-1" or "-4", the "-2" or "-3" cases will be caught by the c# code before even hitting the stored procedure. Since we immediately return for -1/-4, we should have a delegate from the get by candidate number.

Reply:
I don't see where -2 and -3 are caught...?

Reply:
-2 is missing parameters `len(@FirstName) = 0 or len(@LastName) = 0 or @JobGroupID < 1 or @JobGroupID > (SELECT     MAX(JobGroupID) AS JobGroupID FROM JobGroups)`

-3 is existing aliasId, which we aren't setting here